dotCMS/src/main/java/com/dotcms/publisher/util/dependencies/PushPublishigDependencyProcesor.java

```diff
@@ -0,0 +1,796 @@
+                    dependencyProcessor.addAsset(liveTemplateLP, PusheableAsset.TEMPLATE);
+                }
+            }
+
+            // Contents dependencies
+            tryToAddAllAndProcessDependencies(PusheableAsset.CONTENTLET,
+                    pushPublishigDependencyProvider.getContentletsByPage(workingPage));
+
+
+            // Rule dependencies
+            tryToAddAllAndProcessDependencies(PusheableAsset.RULE,
+                    pushPublishigDependencyProvider.getRuleByPage(workingPage));
+        } catch (DotSecurityException | DotDataException e) {
+            Logger.error(this, e.getMessage(),e);
+        }
+    }
+
+    @Override
+    public void addAsset(Object asset, PusheableAsset pusheableAsset) {
+        dependencyProcessor.addAsset(asset, pusheableAsset);
+    }
+
+    @Override
+    public void waitUntilResolveAllDependencies() throws ExecutionException {
+        dependencyProcessor.waitUntilResolveAllDependencies();
+    }
+
+    private <T> boolean add(final PusheableAsset pusheableAsset, final T asset) {
+        final boolean isAdded = config.add(asset, pusheableAsset);
+
+        if (isAdded) {
+            pushedAssetUtil.savePushedAssetForAllEnv(asset, pusheableAsset);
+        }
+
+        return isAdded;
+    }
+
+    private <T> void tryToAddAllAndProcessDependencies(
+            final PusheableAsset pusheableAsset, final Collection<T> assets)
+            throws DotDataException, DotSecurityException {
+
+        assets.stream().forEach(asset -> {
+            try {
+                tryToAdd(pusheableAsset, asset);
+                dependencyProcessor.addAsset(asset, pusheableAsset);
+            } catch (AssetExcludeByFilterException e) {
+                //ignore
+            } catch (AssetExcludeException e) {
+                dependencyProcessor.addAsset(asset, pusheableAsset);
+            }
+        });
+    }
+
+    private <T> Collection<T> tryToAddAll(
+            final PusheableAsset pusheableAsset, final Collection<T> assets)
+            throws DotDataException, DotSecurityException {
+
+        if (assets != null) {
+            return assets.stream()
+                    .filter(asset -> {
+                        try {
+                            return tryToAdd(pusheableAsset, asset);
+                        } catch (AssetExcludeException e) {
+                            return false;
+                        }
+                    })
+                    .collect(Collectors.toSet());
+        } else {
+            return Collections.emptySet();
+        }
+    }
+
+    private <T> void tryToAddAndProcessDependencies(final PusheableAsset pusheableAsset, final T asset)
+            throws DotDataException, DotSecurityException {
+        if (UtilMethods.isSet(asset)) {
+            try {
+                tryToAdd(pusheableAsset, asset);
+                config.addWithDependencies(asset, pusheableAsset);
+            } catch (AssetExcludeByFilterException e) {
```

Comment:
We should try to avoid this exception-based flow control and return a response object if needed.

Reply:
I am going to do it in the next PR